webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -10,36 +10,178 @@ import FormRowCentered from "../components/FormRowCentered";
 import StepForm from "../components/StepForm";
 import StepHeaderButtons from "../components/StepHeaderButtons";
 import SubHeading from "../components/SubHeading";
+import {
+  checkboxPropType,
+  fieldPropType,
+  selectionFieldPropType,
+} from "../lib/propTypes";
 
-const TAX_NUMBER_FORM_STATES = {
+const TAX_NUMBER_FORM_STATE_CHANGES = {
   NothingSelected: "NothingSelected",
-  StateSelectionTaxNumberExists: "StateSelectionTaxNumberExists",
-  StateSelectionNoTaxNumberExists: "StateSelectionNoTaxNumberExists",
-  TaxOfficeSelection: "TaxOfficeSelection",
-  TaxNumberInput: "TaxNumberInput",
-  RequestNewTaxNumber: "RequestNewTaxNumber",
+  TaxNumberExistsSelected: "TaxNumberExistsSelected",
+  BundeslandSelected: "BundeslandSelected",
+  TaxOfficeSelected: "TaxOfficeSelected",
 };
 
-function currentState(taxNumberFormState, fields) {
-  if (taxNumberFormState === TAX_NUMBER_FORM_STATES.NothingSelected) {
-    if (fields.steuernummerExists.value === "yes") {
-      if (fields.bundesland.selectedValue) {
-        return TAX_NUMBER_FORM_STATES.TaxNumberInput;
+function reduceTaxNumberPageData(state, action) {
+  switch (action.type) {
+    case TAX_NUMBER_FORM_STATE_CHANGES.TaxNumberExistsSelected:
+      return {
+        ...state,
+        taxNumberExists: action.value,
+        steuernummer:
+          action.value === "yes" && state.bundesland
+            ? state.steuernummer
+            : [""],
+        bufaNr:
+          action.value !== "yes" && state.bundesland ? state.bufaNr : undefined,
+        requestNewTaxNumber: undefined,
+        showTaxNumberExists: true,
+        showBundesland: true,
+        showSteuernummer: action.value === "yes" && state.bundesland,
+        showBufaNr: action.value !== "yes" && state.bundesland,
+        showRequestNewTaxNumber: false,
+      };
+    case TAX_NUMBER_FORM_STATE_CHANGES.BundeslandSelected:
+      if (!action.value) {
```

Comment:
Interesting. So this is also the case if I select the default value? That is treated as null?

Reply:
Exactly. That's the case